pkg/secret/secrets.go

```diff
@@ -23,3 +25,24 @@ func LoadSecretData(apiReader client.Reader, secretName, namespace, dataKey stri
 	}
 	return string(retStr), nil
 }
+
+func ReadBasicAuthSecret(apiReader clientv1.SecretInterface, secretName string) (string, string, error) {
```

Comment:
Should we also validate that the BasicAuth secret is actually of the basicauth type? Or would that be unnecessarily strict?

Reply:
I personally think that is unnecessarily strict. It shouldn't strictly matter what type of secret it is, as long as it contains both a `username` and a `password` field